tests/acceptance/cli_test.py

```diff
@@ -1320,6 +1321,31 @@ def test_exec_custom_user(self):
         self.assertEqual(stdout, "operator\n")
         self.assertEqual(stderr, "")
 
+    @v2_2_only()
+    def test_exec_service_with_environment_overridden(self):
+        name = 'service'
+        self.base_dir = 'tests/fixtures/environment-exec'
+        self.dispatch(['up', '-d'])
+        self.assertEqual(len(self.project.containers()), 1)
+
+        stdout, stderr = self.dispatch([
+            'exec',
+            '-T',
+            '-e', 'foo=notbar',
+            '--env', 'alpha=beta',
+            name,
+            'env',
+        ])
+
+        # env overridden
+        self.assertIn('foo=notbar', stdout)
```

Comment:
nit: we can use `assert 'foo=notbar' in stdout` instead (and similarly for the lines below), which looks a bit nicer.